src/components/Pages/PlaylistCollection/Container.js

```diff
@@ -45,14 +45,16 @@ const PlaylistCollectionContainer = ({ cId, t, cuId }) => {
 
       // Fetch full units data if needed.
       if (Array.isArray(cuIDs) && cuIDs.length > 0) {
-        cuIDs.forEach(cuID => {
-          if (!fullUnitFetchedMap[cuID] && !wipMap.units[cuID] && !errorMap.units[cuID]) {
-            const cu = content_units.find(x => x.id === cuID);
-            if (!cu || !cu.files || !cu.tags || !cu.sources) {
-              dispatch(actions.fetchUnit(cuID));
-            }
-          }
+        const cusForFetch = cuIDs.filter(cuID => {
+          if (fullUnitFetchedMap[cuID] || wipMap.units[cuID] || errorMap.units[cuID])
+            return false;
+          const cu = content_units.find(x => x.id === cuID);
+          return !cu?.files;
         });
+
+        if (cusForFetch?.length > 0) {
+          dispatch(actions.fetchUnitsByIDs({ id: cusForFetch, with_tags: true, with_files: true }));
```

Comment:
If this is a full fetch, you need to fetch also with_sources

Reply:
but here I dont need sources, I think that can remove tags too. What you think?